mx.py

```diff
@@ -3590,6 +3590,8 @@ def update(self, vcdir, rev=None, mayPull=False, clean=False, abortOnError=False
         """
         if rev and mayPull and not self.exists(vcdir, rev):
             self.pull(vcdir, rev=rev, update=False, abortOnError=abortOnError)
+            if not self.exists(vcdir, rev):
+                abort('Fetch of %s succeeded\nbut did not contain requested revision %s.\nCheck that your remotes agree with the suite.py reposity location' % (vcdir, rev))
```

Comment:
It helps those who do not know what checking remotes against suite.py means or how to do it. Since mx is trying to abstract over VC operations in general, error messages specifically mentioning git concepts and actions can be confusing to the uninitiated. At the very least, the error message can show the git command for checking remotes as well as the full path to the suite.py to check.